packages/angular-table/src/index.ts

```diff
@@ -1,96 +1,72 @@
-  const injector = inject(Injector)
+): Table<TData> {
+  return lazyInit(() => {
+    const resolvedOptions = {
+      state: {},
+      onStateChange: () => {},
+      renderFallbackValue: null,
+      ...options(),
+    }
 
-  return lazyInit(() =>
-    runInInjectionContext(injector, () => {
-      const resolvedOptionsSignal = computed<TableOptionsResolved<TData>>(
-        () => {
-          return {
-            state: {},
-            onStateChange: () => {},
-            renderFallbackValue: null,
-            ...options(),
-          }
-        }
-      )
+    const table = createTable(resolvedOptions)
 
-      const notifier = signal([], { equal: () => false })
-      const table = createTable(untracked(resolvedOptionsSignal))
-      const state = signal(table.initialState)
+    // By default, manage table state here using the table's initial state
+    const state = signal<TableState>(table.initialState)
 
-      function updateOptions() {
-        const tableState = untracked(state)
-        const resolvedOptions = untracked(resolvedOptionsSignal)
-        untracked(() => {
-          table.setOptions(prev => ({
-            ...prev,
-            ...resolvedOptions,
-            state: { ...tableState, ...resolvedOptions.state },
-            onStateChange: updater => {
-              const value =
-                updater instanceof Function ? updater(tableState) : updater
-              state.set(value)
-              resolvedOptions.onStateChange?.(updater)
-            },
-          }))
-        })
+    // Compose table options using computed.
+    // This is to allow `tableSignal` to listen and set table option
+    const updatedOptions = computed<TableOptionsResolved<TData>>(() => {
+      // listen to table state changed
+      const tableState = state()
+      // listen to input options changed
+      const tableOptions = options()
+      return {
+        ...table.options,
+        ...resolvedOptions,
+        ...tableOptions,
+        state: { ...tableState, ...tableOptions.state },
+        onStateChange: updater => {
+          const value =
+            updater instanceof Function ? updater(tableState) : updater
+          state.set(value)
+          resolvedOptions.onStateChange?.(updater)
+        },
       }
+    })
 
-      updateOptions()
-
-      let firstRender = true
-      effect(() => {
-        void [state(), resolvedOptionsSignal()]
-        if (firstRender) {
-          return (firstRender = false)
-        }
-        untracked(() => {
-          updateOptions()
-          notifier.set([])
-        })
-      })
-
-      const tableSignal = computed(
-        () => {
-          notifier()
-          return table
-        },
-        {
-          equal: () => false,
-        }
-      )
+    // convert table instance to signal for proxify to listen to any table state and options changes
+    const tableSignal = computed(
+      () => {
+        table.setOptions(updatedOptions())
```

Comment:
I’d go back with the previous implementation, updating the options inside a signal is theoretically wrong

Reply:
I disagree. Doing side effects not related to value being returned by the computed signal is theoritically wrong. But updating the object before it will be return inside a computed is not. IMHO

Reply:
I wanted to get rid of the `effect` from the start because of its high requirements. But was in doubt because I was not so sure if  I'm doing it wrong to call `updateOptions` method. Adding a `updatedOptions` signal simplifies that and more obvious that is is not wrong to update the Table object before it will returned in the computed signal. Calling `updateOptions` functions seems wrong because we cannot control if someone modifies that method and do something more than updating the table options.